encoder/generator.go

```diff
@@ -30,6 +32,13 @@ const (
 	tgtEnum      attrTSTarget = "enum"
 )
 
+type enumValCofig int
```

Comment:
Why a type? Looking at the logic below, seems like just tracking this with a plain boolean variable should be sufficient

Reply:
i don't have strong opinion on this, it might be more evident if i put the name of the case for people who read it. Let's put the boolean instead.